File: C++/프로그래머스/0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.cpp
#include <vector>
#include <cstdlib>

using namespace std;

int solution(vector<vector<int>> dots) {
    int width = dots[0][0];
    
    for(const auto d: dots) {
        if (width != d[0]) {
            width -= d[0];
            break;
        }
    }
    
    int height = dots[0][1];
    
    for (const auto d: dots) {
        if (height != d[1]) {
            height -= d[1];
            break;
        }
    }
    
    return abs(width * height);
}

File: Go/프로그래머스/0/120860. 직사각형 넓이 구하기/직사각형 넓이 구하기.go
func AbsInt(n int) int {
    if n < 0 {
        return -n
    }
    return n
}

func solution(dots [][]int) int {
    w, h := dots[0][0], dots[0][1]
    
    
    if w != dots[1][0] {
        w = AbsInt(w - dots[1][0])
    } else {
        w = AbsInt(w - dots[2][0])
    }
    
    if h != dots[1][1] {
        h = AbsInt(h - dots[1][1])
    } else {
        h = AbsInt(h - dots[2][1])
    }
    
    return w * h
}